subcommand/server-acl-init/connect_inject.go

```diff
@@ -0,0 +1,182 @@
+			return err
+		})
+	if err != nil {
+		return err
+	}
+	// If Consul namespaces are enabled, someone may have changed their namespace
+	// configuration, so auth methods and binding rules should be updated accordingly
+	if len(existingRules) > 0 && !c.flagEnableNamespaces {
+		c.Log.Info(fmt.Sprintf("Binding rule for %s already exists", authMethodName))
+		return nil
+	}
+
+	var kubeSvc *apiv1.Service
+	err = c.untilSucceeds("getting kubernetes service IP",
+		func() error {
+			var err error
+			kubeSvc, err = c.clientset.CoreV1().Services("default").Get("kubernetes", metav1.GetOptions{})
+			return err
+		})
+	if err != nil {
+		return err
+	}
+
+	// Get the Secret name for the auth method ServiceAccount.
+	var authMethodServiceAccount *apiv1.ServiceAccount
+	saName := c.withPrefix("connect-injector-authmethod-svc-account")
+	err = c.untilSucceeds(fmt.Sprintf("getting %s ServiceAccount", saName),
+		func() error {
+			var err error
+			authMethodServiceAccount, err = c.clientset.CoreV1().ServiceAccounts(c.flagK8sNamespace).Get(saName, metav1.GetOptions{})
+			return err
+		})
+	if err != nil {
+		return err
+	}
+
+	// ServiceAccounts always have a secret name. The secret
+	// contains the JWT token.
+	saSecretName := authMethodServiceAccount.Secrets[0].Name
+
+	// Get the secret that will contain the ServiceAccount JWT token.
+	var saSecret *apiv1.Secret
+	err = c.untilSucceeds(fmt.Sprintf("getting %s Secret", saSecretName),
+		func() error {
+			var err error
+			saSecret, err = c.clientset.CoreV1().Secrets(c.flagK8sNamespace).Get(saSecretName, metav1.GetOptions{})
+			return err
+		})
+	if err != nil {
+		return err
+	}
+
+	// Now we're ready to set up Consul's auth method.
+	authMethodTmpl := api.ACLAuthMethod{
+		Name:        authMethodName,
+		Description: "Kubernetes AuthMethod",
+		Type:        "kubernetes",
+		Config: map[string]interface{}{
+			"Host":              fmt.Sprintf("https://%s:443", kubeSvc.Spec.ClusterIP),
+			"CACert":            string(saSecret.Data["ca.crt"]),
+			"ServiceAccountJWT": string(saSecret.Data["token"]),
+		},
+	}
+
+	// Add options for mirroring namespaces
+	if c.flagEnableInjectK8SNSMirroring {
+		authMethodTmpl.Config["MapNamespaces"] = true
+		authMethodTmpl.Config["ConsulNamespacePrefix"] = c.flagInjectK8SNSMirroringPrefix
+	}
+
+	// Set up the auth method in the specific namespace if not mirroring
+	// If namespaces and mirroring are enabled, this is not necessary because
+	// the auth method will fall back to being created in the Consul `default`
+	// namespace automatically, as is necessary for mirroring.
+	// Note: if the config changes, an auth method will be created in the
+	// correct namespace, but the old auth method will not be removed.
+	writeOptions := api.WriteOptions{}
+	if c.flagEnableNamespaces && !c.flagEnableInjectK8SNSMirroring {
+		writeOptions.Namespace = c.flagConsulInjectDestinationNamespace
+
+		if c.flagConsulInjectDestinationNamespace != "default" {
```

Comment:
If this showed up a number of places, I would be happy to. However, this is the only place where it is used. It doesn't justify the switch, especially since we don't do the same for the k8s namespace. It is also highly unlikely to ever change, either in Consul or k8s.